src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Marshaller;
+
+/**
+ * Serializes/unserializes values using igbinary_serialize() if available, serialize() otherwise.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class DefaultMarshaller implements MarshallerInterface
+{
+    private $allowIgbinarySerialize = true;
+
+    public function __construct(bool $allowIgbinarySerialize = true)
```

Comment:
A way to force only igbinary (disable the PHP serialization fallback)?

Reply:
@palex-fpt said:
> I prefer fast fail when igbinary is configured for application but is not available on deployment node, than slow running application with hard to detect reasons.

To which @nicolas-grekas you replied:
> No way: that explicitly conflicts with PSR-6 spirit and wordings (for good reasons IMHO.)

But if the fail-fast behavior means a cache miss, that's not violating PSR-6 in any way.

Reply:
Correct. Not convinced yet. Really worth it?

Reply:
TBH, I prefer fail-fast feature over auto-adaptive feature. Reconfigured application cache can and should be warmed up. But having spoiled node that writes to shared cache in php format and forces all other properly configured nodes to use slow method is not good. If someone concerning about preserving old cache data he can use workaround with distinct namespaces and chain cache instances with igbinary an php-serialize.

Reply:
Here we are: the default mode is auto-adaptative, but if you pass `true`, you'll get an exception when igbinary is not loaded.